crates/compiler/test_syntax/src/test_helpers.rs

```diff
@@ -116,6 +133,81 @@ impl<'a> Output<'a> {
             Output::Full { .. } => format!("{self:#?}\n"),
         }
     }
+
+    pub fn canonicalize(&self, arena: &Bump, src: &str) {
+        set_panic_not_exit(true); // can has a bunch of internal_error! calls
+
+        match self {
+            Output::Header(_) => {}
+            Output::ModuleDefs(_) => {
+                // TODO: canonicalize module defs
+            }
+            Output::Full(_) => {
+                // TODO: canonicalize full ast
```

Comment:
TODO

Reply:
(same as above; will implement this eventually - after fixing the current panics)